### PR TITLE
Make examples use .schema.json file

### PR DESCRIPTION
**Pull request by bennetthardwick**
- Convert examples to use JSON schema file
- Add example for using TypeScript schema dsl

### CODE REVIEW

File: example-schema.ts [hunk 1]
```diff
@@ -12,22 +12,20 @@ export type Movie = {
...
+  const moviesSchema = await new Promise<string>((resolve, reject) => {
+    readFile(join(__dirname, "./movies.schema.json"), (error, buffer) => {
+      if (error) {
+        return reject(error);
+      }
+
+      resolve(buffer.toString());
+    });
+  });
+
```
Comment (coderdan) [inline example-schema.ts]: Isn't this step redundant given that the schema is stored in the collection itself? I'm confused.

**Comment by freshtonic** [comment]
Where are all the semicolons coming from? Are they from Prettier?

CipherStash's TS code should be a semicolon-free zone :)

File: example-schema.ts (hunk 1 above)
Comment (coderdan) [reply example-schema.ts]: I liked the ability to do `stash.loadCollection("movies")` - is that possible?

**Review by freshtonic — APPROVED** [review]
Looks good (apart from the semicolons!)

File: example-schema.ts (hunk 1 above)
Comment (bennetthardwick) [reply example-schema.ts]: Good point, I suppose the only example that actually needs this is the create collection one. I'll update everything.

**Review by coderdan — DISMISSED** [review]
I'm keen to understand the thinking behind this approach.

**Comment by bennetthardwick** [comment]
> Where are all the semicolons coming from? Are they from Prettier?
> 
> CipherStash's TS code should be a semicolon-free zone :)

Yep, probably coming from prettier or muscle memory. I'll remove and then add a prettierrc in another PR 👍🏻

**Review by coderdan — APPROVED** [review]

**Review by freshtonic — APPROVED** [review]

**Pull request merged**